pkg/backend/kubeflow/generator/generator.go

```diff
@@ -0,0 +1,13 @@
+package generator
+
+import (
+	pytorchv1alpha2 "github.com/kubeflow/pytorch-operator/pkg/apis/pytorch/v1alpha2"
+	tfv1alpha2 "github.com/kubeflow/tf-operator/pkg/apis/tensorflow/v1alpha2"
+
+	"github.com/caicloud/ciao/pkg/types"
+)
+
+type Interface interface {
+	GenerateTFJob(parameter *types.Parameter) *tfv1alpha2.TFJob
+	GeneratePyTorchJob(parameter *types.Parameter) *pytorchv1alpha2.PyTorchJob
```

Comment:
It seems generator should somehow be versioned, since you are generating versioned types.